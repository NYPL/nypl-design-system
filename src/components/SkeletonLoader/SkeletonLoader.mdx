import { ArgTypes, Canvas, Description, Meta } from "@storybook/blocks";

import * as SkeletonLoaderStories from "./SkeletonLoader.stories";

<Meta of={SkeletonLoaderStories} />

# SkeletonLoader

| Component Version | DS Version |
| ----------------- | ---------- |
| Added             | `0.17.3`   |
| Latest            | `1.5.0`    |

## Table of Contents

- [Overview](#overview)
- [Component Props](#component-props)
- [Accessibility](#accessibility)
- [Card Placeholders in a Grid](#card-placeholders-in-a-grid)
- [Text Placeholders in a List](#text-placeholders-in-a-list)

## Overview

<Description of={SkeletonLoaderStories} />

The `layout`, `width`, and placeholder elements within the `SkeletonLoader` can
be customized.

Please note that the initial rendering below has the `width` prop set to `300px`
to better view the example. The default value is `100%`.

## Component Props

<Canvas of={SkeletonLoaderStories.WithControls} />

<ArgTypes of={SkeletonLoaderStories.WithControls} />

## Accessibility

According to W3C's success criterion 2.2.2 for operable user interfaces:

> For any moving, blinking or scrolling information that (1) starts
> automatically, (2) lasts more than five seconds, and (3) is presented in
> parallel with other content, there is a mechanism for the user to pause, stop,
> or hide it unless the movement, blinking, or scrolling is part of an activity
> where it is essential;

The `SkeletonLoader` component can be used as a placeholder for local content
that is loaded in parallel but it typically does not last more than five (5)
seconds. For the DS usage, this is purely a visual cue and should not be picked
up by screen readers.

Resources:

- [W3C Perceivable Pause, Stop, Hide](https://www.w3.org/TR/WCAG/#pause-stop-hide)
- [More Accessible Skeletons](https://adrianroselli.com/2020/11/more-accessible-skeletons.html)

## Card Placeholders in a Grid

<Canvas of={SkeletonLoaderStories.GridExample} />

## Text Placeholders in a List

<Canvas of={SkeletonLoaderStories.ListExample} />
