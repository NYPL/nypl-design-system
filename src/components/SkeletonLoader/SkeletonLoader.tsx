import React from "react";

function SkeletonLoader() {
  return <div className="skeleton-loader"></div>;
}

export default SkeletonLoader;
