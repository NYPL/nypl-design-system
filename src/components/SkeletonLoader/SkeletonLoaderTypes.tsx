export enum SkeletonLoaderLayouts {
  Horizontal = "horizontal",
  Vertical = "vertical",
}

export enum SkeletonLoaderImageRatios {
  Landscape = "landscape",
  Portrait = "portrait",
  Square = "square",
}
