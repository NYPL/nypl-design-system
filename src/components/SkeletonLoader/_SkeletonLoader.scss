.skeleton-loader:empty {
  animation: shine 1s infinite;
  background-image: linear-gradient(
      100deg,
      rgba(255, 255, 255, 0),
      rgba(255, 255, 255, 0.5) 50%,
      rgba(255, 255, 255, 0) 80%
    ),
    linear-gradient(var(--ui-gray-warm-light) 33px, transparent 0),
    linear-gradient(var(--ui-gray-warm-light) 20px, transparent 0),
    linear-gradient(var(--ui-gray-warm-light) 20px, transparent 0),
    linear-gradient(var(--ui-gray-warm-light) 20px, transparent 0),
    linear-gradient(var(--ui-gray-warm-light) 20px, transparent 0),
    linear-gradient(var(--ui-gray-warm-light) 20px, transparent 0),
    linear-gradient(var(--ui-gray-warm-light) 20px, transparent 0);
  background-position: 0 0, /* highlight */ 0px 0, 0px 40px, 0px 67px, 0px 94px,
    0px 121px, 0px 148px, 0px 175px;
  background-repeat: repeat-y;
  background-size: 50px 230px, /* highlight */ 300px 230px, 250px 230px,
    250px 230px, 250px 230px, 200px 230px, 200px 230px, 200px 230px;
  height: 485px; /* change height to see repeat-y behavior */
  margin: auto;
  width: 100%;
}

@keyframes shine {
  to {
    background-position: 100% 0, /* move highlight to right */ 0px 0, 0px 40px,
      0px 67px, 0px 94px, 0px 121px, 0px 148px, 0px 175px;
  }
}
