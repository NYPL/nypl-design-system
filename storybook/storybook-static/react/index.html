<!doctype html><html lang="en"><head><meta charset="utf-8"><title>Storybook</title><meta name="viewport" content="width=device-width,initial-scale=1"><style>html, body {
    overflow: hidden;
    height: 100%;
    width: 100%;
    margin: 0;
    padding: 0;
  }</style><script>/* globals window */
  /* eslint-disable no-underscore-dangle */
  try {
    if (window.top !== window) {
      window.__REACT_DEVTOOLS_GLOBAL_HOOK__ = window.top.__REACT_DEVTOOLS_GLOBAL_HOOK__;
    }
  } catch (e) {
    // eslint-disable-next-line no-console
    console.warn('unable to connect to top frame for connecting dev tools');
  }</script><style>#root[hidden],
      #docs-root[hidden] {
        display: none !important;
      }</style></head><body><div id="root"></div><div id="docs-root"></div><script>window['DOCS_MODE'] = false;</script><script src="runtime~main.00bd9d033d7a0b972fc9.bundle.js"></script><script src="vendors~main.d2b6aafef1a259d165c0.bundle.js"></script><script src="main.3bf71672a6dd35499004.bundle.js"></script></body></html>