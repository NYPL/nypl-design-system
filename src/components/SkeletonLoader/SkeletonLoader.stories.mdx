import {
  Meta,
  Story,
  Canvas,
  ArgsTable,
  Preview,
  Description,
} from "@storybook/addon-docs/blocks";
import { withDesign } from "storybook-addon-designs";

import SkeletonLoader from "./SkeletonLoader";
import { action } from "@storybook/addon-actions";

<Meta
  title="Skeleton Loader"
  component={SkeletonLoader}
  decorators={[withDesign]}
  parameters={{
    design: {
      type: "figma",
      url:
        "https://www.figma.com/file/qShodlfNCJHb8n03IFyApM/Master?node-id=17219%3A7",
    },
  }}
  argTypes={{
    attributes: { table: { disable: true } },
    className: { table: { disable: true } },
    modifiers: { table: { disable: true } },
    layout: {
      table: { defaultValue: { summary: "vertical" } },
    },
    showImage: {
      table: { defaultValue: { summary: "true" } },
    },
    showHeading: {
      table: { defaultValue: { summary: "true" } },
    },
    showContent: {
      table: { defaultValue: { summary: "true" } },
    },
    showButton: {
      table: { defaultValue: { summary: "false" } },
    },
    headingSize: {
      table: { defaultValue: { summary: "1" } },
    },
    contentSize: {
      table: { defaultValue: { summary: "3" } },
    },
    imageAspectRatio: {
      table: { defaultValue: { summary: "square" } },
    },
  }}
/>

# Skeleton Loader

<Description of={SkeletonLoader} />

The `SkeletonLoader` component renders a placeholder to be used while dynamic content is loading. The `layout`, `width` and placeholder elements within the `SkeletonLoader` can be customized.

Please note that the initial rendering below has the `width` prop set to `300px`. The default value is `100%`.

<Preview withToolbar>
  <Story
    name="Skeleton Loader"
    args={{
      width: "300px",
    }}
  >
    {args => <SkeletonLoader {...args} />}
  </Story>
</Preview>

<ArgsTable story="Skeleton Loader" />
