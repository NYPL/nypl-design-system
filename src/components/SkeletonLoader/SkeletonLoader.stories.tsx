import * as React from "react";

import SkeletonLoader from "./SkeletonLoader";
import { withDesign } from "storybook-addon-designs";

export default {
  title: "Skeleton Loader",
  component: SkeletonLoader,
  decorators: [withDesign],
};

export const skeletonLoader = () => <SkeletonLoader />;

skeletonLoader.storyName = "Skeleton Loader";
skeletonLoader.parameters = {
  design: {
    type: "figma",
    url:
      "https://www.figma.com/file/qShodlfNCJHb8n03IFyApM/Master?node-id=17219%3A7",
  },
};
