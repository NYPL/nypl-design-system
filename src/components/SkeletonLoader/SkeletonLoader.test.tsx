import * as React from "react";
import { render } from "@testing-library/react";
import { axe } from "jest-axe";
import renderer from "react-test-renderer";

import SkeletonLoader from "./SkeletonLoader";

describe("SkeletonLoader Accessibility", () => {
  it("passes axe accessibility test", async () => {
    const { container } = render(<SkeletonLoader />);
    expect(await axe(container)).toHaveNoViolations();
  });
});

describe("SkeletonLoader", () => {
  it("renders default layout", () => {
    const { container } = render(
      <SkeletonLoader className="skeleton-loader" />
    );

    expect(container.querySelector(".skeleton-loader")).toBeInTheDocument();
  });

  it("renders in the column or row layout", () => {
    const { container, rerender } = render(<SkeletonLoader layout="column" />);

    expect(container.querySelector(".column")).toBeInTheDocument();

    rerender(<SkeletonLoader layout="row" />);
    expect(container.querySelector(".row")).toBeInTheDocument();
  });

  it("renders default elements", () => {
    const { container } = render(<SkeletonLoader />);

    // By default, the `SkeletonLoader` renders one image, one heading, and
    // three content elements.
    expect(container.querySelectorAll(".chakra-skeleton")).toHaveLength(5);
  });

  it("renders without image", () => {
    const { container } = render(<SkeletonLoader showImage={false} />);

    // Only one image is rendered by default.
    expect(container.querySelectorAll(".chakra-skeleton")).toHaveLength(4);
  });

  it("renders without heading", () => {
    const { container } = render(<SkeletonLoader showHeading={false} />);

    // Only one heading is rendered by default.
    expect(container.querySelectorAll(".chakra-skeleton")).toHaveLength(4);
  });

  it("renders without content", () => {
    const { container } = render(<SkeletonLoader showContent={false} />);

    // Three content placeholders are rendered by default.
    expect(container.querySelectorAll(".chakra-skeleton")).toHaveLength(2);
  });

  it("renders with button", () => {
    const { container } = render(<SkeletonLoader showButton={true} />);

    // Only one button is rendered by default.
    expect(container.querySelectorAll(".chakra-skeleton")).toHaveLength(6);
  });

  it("renders the UI snapshot correctly", () => {
    const basic = renderer.create(<SkeletonLoader />).toJSON();
    const rowLayout = renderer.create(<SkeletonLoader layout="row" />).toJSON();
    const columnLayout = renderer
      .create(<SkeletonLoader layout="column" />)
      .toJSON();
    const noImage = renderer
      .create(<SkeletonLoader showImage={false} />)
      .toJSON();
    const noHeading = renderer
      .create(<SkeletonLoader showHeading={false} />)
      .toJSON();
    const noContent = renderer
      .create(<SkeletonLoader showContent={false} />)
      .toJSON();
    const withButton = renderer
      .create(<SkeletonLoader showButton={true} />)
      .toJSON();
    const landscape = renderer
      .create(<SkeletonLoader imageAspectRatio="landscape" />)
      .toJSON();
    const portrait = renderer
      .create(<SkeletonLoader imageAspectRatio="portrait" />)
      .toJSON();
    const square = renderer
      .create(<SkeletonLoader imageAspectRatio="square" />)
      .toJSON();
    const withChakraProps = renderer
      .create(<SkeletonLoader p="20px" color="ui.error.primary" />)
      .toJSON();
    const withOtherProps = renderer
      .create(<SkeletonLoader data-testid="props" />)
      .toJSON();

    expect(basic).toMatchSnapshot();
    expect(rowLayout).toMatchSnapshot();
    expect(columnLayout).toMatchSnapshot();
    expect(noImage).toMatchSnapshot();
    expect(noHeading).toMatchSnapshot();
    expect(noContent).toMatchSnapshot();
    expect(withButton).toMatchSnapshot();
    expect(landscape).toMatchSnapshot();
    expect(portrait).toMatchSnapshot();
    expect(square).toMatchSnapshot();
    expect(withChakraProps).toMatchSnapshot();
    expect(withOtherProps).toMatchSnapshot();
  });
});
