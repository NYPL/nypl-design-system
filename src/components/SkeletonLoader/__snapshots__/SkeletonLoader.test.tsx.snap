// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkeletonLoader renders the UI snapshot correctly 1`] = `
<div
  className="css-1xdhyk6"
  style={
    Object {
      "width": undefined,
    }
  }
>
  <div
    className="chakra-skeleton css-1twknw3"
  >
    <div
      className="css-0"
    />
  </div>
  <div
    className="column css-0"
  >
    <div
      className="css-0"
    >
      <div
        className="chakra-skeleton css-tz3wol"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": null,
            }
          }
        />
      </div>
    </div>
    <div
      className="css-0"
    >
      <div
        className="chakra-skeleton css-1akrz52"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": null,
            }
          }
        />
      </div>
      <div
        className="chakra-skeleton css-1akrz52"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": null,
            }
          }
        />
      </div>
      <div
        className="chakra-skeleton css-qaeww3"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": "0",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`SkeletonLoader renders the UI snapshot correctly 2`] = `
<div
  className="css-1xdhyk6"
  style={
    Object {
      "width": undefined,
    }
  }
>
  <div
    className="chakra-skeleton css-1twknw3"
  >
    <div
      className="css-0"
    />
  </div>
  <div
    className="row css-0"
  >
    <div
      className="css-0"
    >
      <div
        className="chakra-skeleton css-tz3wol"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": null,
            }
          }
        />
      </div>
    </div>
    <div
      className="css-0"
    >
      <div
        className="chakra-skeleton css-1akrz52"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": null,
            }
          }
        />
      </div>
      <div
        className="chakra-skeleton css-1akrz52"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": null,
            }
          }
        />
      </div>
      <div
        className="chakra-skeleton css-qaeww3"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": "0",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`SkeletonLoader renders the UI snapshot correctly 3`] = `
<div
  className="css-1xdhyk6"
  style={
    Object {
      "width": undefined,
    }
  }
>
  <div
    className="chakra-skeleton css-1twknw3"
  >
    <div
      className="css-0"
    />
  </div>
  <div
    className="column css-0"
  >
    <div
      className="css-0"
    >
      <div
        className="chakra-skeleton css-tz3wol"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": null,
            }
          }
        />
      </div>
    </div>
    <div
      className="css-0"
    >
      <div
        className="chakra-skeleton css-1akrz52"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": null,
            }
          }
        />
      </div>
      <div
        className="chakra-skeleton css-1akrz52"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": null,
            }
          }
        />
      </div>
      <div
        className="chakra-skeleton css-qaeww3"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": "0",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`SkeletonLoader renders the UI snapshot correctly 4`] = `
<div
  className="css-1xdhyk6"
  style={
    Object {
      "width": undefined,
    }
  }
>
  <div
    className="column css-0"
  >
    <div
      className="css-0"
    >
      <div
        className="chakra-skeleton css-tz3wol"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": null,
            }
          }
        />
      </div>
    </div>
    <div
      className="css-0"
    >
      <div
        className="chakra-skeleton css-1akrz52"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": null,
            }
          }
        />
      </div>
      <div
        className="chakra-skeleton css-1akrz52"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": null,
            }
          }
        />
      </div>
      <div
        className="chakra-skeleton css-qaeww3"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": "0",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`SkeletonLoader renders the UI snapshot correctly 5`] = `
<div
  className="css-1xdhyk6"
  style={
    Object {
      "width": undefined,
    }
  }
>
  <div
    className="chakra-skeleton css-1twknw3"
  >
    <div
      className="css-0"
    />
  </div>
  <div
    className="column css-0"
  >
    <div
      className="css-0"
    >
      <div
        className="chakra-skeleton css-1akrz52"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": null,
            }
          }
        />
      </div>
      <div
        className="chakra-skeleton css-1akrz52"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": null,
            }
          }
        />
      </div>
      <div
        className="chakra-skeleton css-qaeww3"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": "0",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`SkeletonLoader renders the UI snapshot correctly 6`] = `
<div
  className="css-1xdhyk6"
  style={
    Object {
      "width": undefined,
    }
  }
>
  <div
    className="chakra-skeleton css-1twknw3"
  >
    <div
      className="css-0"
    />
  </div>
  <div
    className="column css-0"
  >
    <div
      className="css-0"
    >
      <div
        className="chakra-skeleton css-tz3wol"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": null,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`SkeletonLoader renders the UI snapshot correctly 7`] = `
<div
  className="css-1xdhyk6"
  style={
    Object {
      "width": undefined,
    }
  }
>
  <div
    className="chakra-skeleton css-1twknw3"
  >
    <div
      className="css-0"
    />
  </div>
  <div
    className="column css-0"
  >
    <div
      className="css-0"
    >
      <div
        className="chakra-skeleton css-tz3wol"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": null,
            }
          }
        />
      </div>
    </div>
    <div
      className="css-0"
    >
      <div
        className="chakra-skeleton css-1akrz52"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": null,
            }
          }
        />
      </div>
      <div
        className="chakra-skeleton css-1akrz52"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": null,
            }
          }
        />
      </div>
      <div
        className="chakra-skeleton css-qaeww3"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": "0",
            }
          }
        />
      </div>
    </div>
    <div
      className="css-0"
    >
      <div
        className="chakra-skeleton css-uhwxm5"
      >
        <div
          className="css-0"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`SkeletonLoader renders the UI snapshot correctly 8`] = `
<div
  className="css-1xdhyk6"
  style={
    Object {
      "width": undefined,
    }
  }
>
  <div
    className="chakra-skeleton css-1twknw3"
  >
    <div
      className="css-0"
    />
  </div>
  <div
    className="column css-0"
  >
    <div
      className="css-0"
    >
      <div
        className="chakra-skeleton css-tz3wol"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": null,
            }
          }
        />
      </div>
    </div>
    <div
      className="css-0"
    >
      <div
        className="chakra-skeleton css-1akrz52"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": null,
            }
          }
        />
      </div>
      <div
        className="chakra-skeleton css-1akrz52"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": null,
            }
          }
        />
      </div>
      <div
        className="chakra-skeleton css-qaeww3"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": "0",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`SkeletonLoader renders the UI snapshot correctly 9`] = `
<div
  className="css-1xdhyk6"
  style={
    Object {
      "width": undefined,
    }
  }
>
  <div
    className="chakra-skeleton css-1twknw3"
  >
    <div
      className="css-0"
    />
  </div>
  <div
    className="column css-0"
  >
    <div
      className="css-0"
    >
      <div
        className="chakra-skeleton css-tz3wol"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": null,
            }
          }
        />
      </div>
    </div>
    <div
      className="css-0"
    >
      <div
        className="chakra-skeleton css-1akrz52"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": null,
            }
          }
        />
      </div>
      <div
        className="chakra-skeleton css-1akrz52"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": null,
            }
          }
        />
      </div>
      <div
        className="chakra-skeleton css-qaeww3"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": "0",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`SkeletonLoader renders the UI snapshot correctly 10`] = `
<div
  className="css-1xdhyk6"
  style={
    Object {
      "width": undefined,
    }
  }
>
  <div
    className="chakra-skeleton css-1twknw3"
  >
    <div
      className="css-0"
    />
  </div>
  <div
    className="column css-0"
  >
    <div
      className="css-0"
    >
      <div
        className="chakra-skeleton css-tz3wol"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": null,
            }
          }
        />
      </div>
    </div>
    <div
      className="css-0"
    >
      <div
        className="chakra-skeleton css-1akrz52"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": null,
            }
          }
        />
      </div>
      <div
        className="chakra-skeleton css-1akrz52"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": null,
            }
          }
        />
      </div>
      <div
        className="chakra-skeleton css-qaeww3"
      >
        <div
          className="css-0"
          style={
            Object {
              "marginBottom": "0",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
